Assets/Scripts/UI/GalleryMenu.cs

```diff
@@ -0,0 +1,221 @@
+        gridElements.RemoveAt(0); // GetComponentInChildren returns this element as well, which we don't want
+
+        maxPages = Mathf.CeilToInt((float)unlockedElements.Length / gridElements.Count);
+
+        CreateTabs(maxPages);
+        PopulateGrid(0);
+    }
+
+    public void SetPlayerInput(InputManager inputManager)
+    {
+        inputManager.onLeftTab += PrevPage;
+        inputManager.onRightTab += NextPage;
+        inputManager.onCancel += Back;
+    }
+
+    private void CreateTabs(int pages)
+    {
+        for (int i = 1; i <= pages; i++)
+        {
+            TabsButton tab = Instantiate(tabPrefab, navigation.transform).GetComponent<TabsButton>();
+            tab.GetComponentInChildren<TMP_Text>().text = (i).ToString();
+            tabs.Add(tab);
+            tab.GetComponent<Image>().color = disabledColor;
+        }
+
+        selectedTab = tabs[0];
+        selectedTab.GetComponent<Image>().color = selectedColor;
+    }
+
+    /// <summary>
+    /// Update the grid items with the corresponding page of items
+    /// </summary>
+    void PopulateGrid(int page)
+    {
+        // Remove previous guns
+        foreach (GameObject g in spawnedWeapons)
+        {
+            GameObject.DestroyImmediate(g);
+        }
+        spawnedWeapons.Clear();
+
+        FlexibleGridLayout gridLayout = gridBase.GetComponent<FlexibleGridLayout>();
+
+        int unlockedElementsOffset = gridElements.Count * page;
+        Vector2 gridCellSize = new Vector2((gridLayout.cellSize.x - gridLayout.spacing.x) / 2, (gridLayout.cellSize.y - gridLayout.spacing.y) / 2);
+
+        for (int i = unlockedElementsOffset; i < gridElements.Count + unlockedElementsOffset; i++)
+        {
+            RectTransform gridElement = gridElements[i%gridElements.Count];
+            gridElement.gameObject.SetActive(true);
+
+            if (i < unlockedElements.Length)
+            { 
+                Weapon weapon = unlockedElements[i];
+                GameObject gun = GunFactory.InstantiateGun(weapon.body, weapon.barrel, weapon.extension, null);
+                gun.transform.localScale = Vector3.one;
+                gun.transform.SetParent(gridElement);
+
+                spawnedWeapons.Add(gun);
+
+                Bounds bounds = new Bounds();
+                foreach (var renderer in gun.GetComponentsInChildren<Renderer>())
+                {
+                    renderer.gameObject.layer = LayerMask.NameToLayer("UI");
+                    bounds.Encapsulate(renderer.bounds);
+                }
+
+                // Scale the weapon so it fits within the UI 
+                float scaleFactor = gridCellSize.x / bounds.size.z;
+
+                gun.transform.localScale = new Vector3(scaleFactor / gun.transform.lossyScale.x, scaleFactor / gun.transform.lossyScale.y, scaleFactor / gun.transform.lossyScale.z);
+
+                // Recalculate bounds
```

Comment:
Perhaps we should turn this into a method, since you're using it twice here and it's a little convoluted
(also, you're setting the layer in both loops, is that necessary?)